packages/navigation/types/Navigation.js

```diff
@@ -55,25 +55,4 @@ export type RouteNames =
   | HomepageStackNavigatorRouteNames
   | PlaygroundNavigationRouteNames;
 
-export type NavigationListener = (
-  key: 'willBlur' | 'willFocus' | 'didFocus' | 'didBlur',
-  // See payload https://reactnavigation.org/docs/en/navigation-prop.html#addlistener-subscribe-to-updates-to-navigation-lifecycle
-  callback: (payload: Object) => void,
-) => { remove: () => void };
-
-/**
- * @see https://reactnavigation.org/docs/navigators/navigation-prop
- */
-export type Navigation = {
-  navigate: ({
-    routeName: RouteNames,
-    key: string, // should be unique
```

Comment:
We used custom type on purpose so we are forced to use the `key` (and therefore avoid the double opening of each screen on double-tap). This is how I understood it at that time and I am afraid you just removed this check so we are not going to be able to prevent this bug.

Reply:
Ok, first it will be good to add a comment saying about the `key` because we as new in the project, we thought you have custom types because the ones from `react-navigation 1.x.x` were not really great (broken all the time).

What do you think of:

```js
type Navigation = NavigationScreenProp<NavigationStateParameters> & { 
 key: string,
}
```

That is the only custom thing you need here, right? About `NavigationStateParameters`...  Right now in your codebase you have it as:

```js
type NavigationStateParameters = Object;
```

Ideally we would have:

```js
type Navigation<S> = NavigationScreenProp<S> & { 
 key: string,
}
```

But that would require components that use that type.

Reply:
> first it will be good to add a comment saying about the `key`

Add it, please.

About your suggestion - I didn't try it but I get your point. It could work. Not sure if it disallows the string instead of the object though.